### PR TITLE
Feature/nas instance

### PR DESCRIPTION
**Pull request by aokumasan**
## Description

- Add nifcloud_nas_instance resource.

## How Has This Been Tested?

- [x]  Buld with `make install` command, and Apply example.

```:console
make install
cd examples/nas_instance
terraform init -plugin-dir ~/.terraform.d/plugins
terraform plan
terraform apply
```

- [x] Run acceptance tests.

```:console
TF_ACC=1 go test ./nifcloud/acc/... -v -count 1 -parallel 20 -timeout 360m -run TestAcc_NASInstance
```

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: nifcloud/resources/nas/nasinstance/flattener_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		{
+			name: "flattens the response",
+			args: args{
+				d: rd,
+				res: &nas.DescribeNASInstancesResponse{
+					DescribeNASInstancesOutput: &nas.DescribeNASInstancesOutput{
+						NASInstances: []nas.NASInstance{
+							{
+								NASInstanceIdentifier:  nifcloud.String("test_identifier"),
+								AllocatedStorage:       nifcloud.String("100"),
```
Comment (tunakyonn) [inline nifcloud/resources/nas/nasinstance/flattener_test.go:56]:
```suggestion
								AllocatedStorage:       nifcloud.Int64(100),
```

Comment (aokumasan) [reply nifcloud/resources/nas/nasinstance/flattener_test.go:56]: the type of AllocatedStorage in DescribeNASInstances response is `*string` 😇 

https://github.com/nifcloud/nifcloud-sdk-go/blob/v1.5.0/service/nas/api_types.go#L72

File: nifcloud/resources/nas/nasinstance/schema.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+			Type:        schema.TypeString,
+			Description: "The AZ for the NAS instance.",
+			Optional:    true,
+			Computed:    true,
+			ForceNew:    true,
+		},
+		"identifier": {
+			Type:        schema.TypeString,
+			Description: "The name of the NAS instance.",
+			Required:    true,
```
[the remark this answers is not in the document]
Comment (aokumasan) [reply nifcloud/resources/nas/nasinstance/schema.go]: Thanks!!! applied!

File: nifcloud/resources/nas/nasinstance/flattener_test.go (hunk 1 above)
Comment (tunakyonn) [reply nifcloud/resources/nas/nasinstance/flattener_test.go:56]: Oh, sorry 😢

**Review by tunakyonn — APPROVED** [review]
LGTM!

**Pull request merged**